[SQL] User-defined aggregate generates uncompilable Rust code

```sql
CREATE LINEAR AGGREGATE u256_sum(value BINARY(32)) RETURNS BINARY(32);

CREATE TABLE A (
    id VARCHAR(20) NOT NULL PRIMARY KEY,
    a VARCHAR(64),
    b VARCHAR(64),
    sno INT,
    small BIGINT, 
    num1 BINARY(32),
    num2 BINARY(32)
) WITH ('append_only' = 'true');

CREATE VIEW b AS 
SELECT id, a, b, u256_sum(num1), u256_sum(num2), SUM(small), MAX(sno)
FROM A
GROUP BY id, a, b
```

```
error[E0507]: cannot move out of `*p0.0` which is behind a shared reference
  --> crates/feldera_pipe_operator_884c63bdeca88dc9/src/lib.rs:72:152
   |
72 | ...Option<ByteArray>, Option<i64>, Option<i32>>, ) = ((*p0.0), Tup4::new(
   |                                                       ^^^^^^^ move occurs because `*p0.0` has type `Tup3<SqlString, Option<SqlString>, Option<SqlString>>`, which does not implement the `Copy` trait
   |
   = note: the full name for the type has been written to '/home/ubuntu/.feldera/compiler/rust-compilation/target/optimized/deps/feldera_pipe_operator_884c63bdeca88dc9-2f3e95d813909fdb.long-type-3112468952754669319.txt'
   = note: consider using `--verbose` to print the full type name to the console
```

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/outer/ExpandAggregates.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/outer/ExpandAggregates.java
@@ -247,7 +247,7 @@ public class ExpandAggregates extends Passes {
             if (!ga.shuffle.isIdentityPermutation()) {
                 DBSPTypeIndexedZSet ix = result.getOutputIndexedZSetType();
                 DBSPVariablePath vReorder = ix.getKVRefType().var();
-                DBSPTupleExpression key = DBSPTupleExpression.flatten(vReorder.field(0).deref());
+                DBSPTupleExpression key = DBSPTupleExpression.flatten(vReorder.field(0).deref().applyCloneIfNeeded());
                 Shuffle inverse = ga.shuffle.invert();
                 DBSPBaseTupleExpression value =
                         DBSPTupleExpression.flatten(vReorder.field(1).deref())
@@ -365,7 +365,7 @@ public class ExpandAggregates extends Passes {
                 // If all operations are MIN or MAX no postprocessing is necessary.
                 this.addOperator(chain);
                 DBSPClosureExpression post = new DBSPRawTupleExpression(
-                        postVar.field(0).deref(),
+                        postVar.field(0).deref().applyCloneIfNeeded(),
                         new DBSPTupleExpression(postProcessing, false))
                         .closure(postVar);
                 chain = new DBSPMapIndexOperator(node, post, chain.outputPort());

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java
@@ -707,6 +707,53 @@ public class MetadataTests extends BaseSQLTests {
     }
 
     @Test
+    public void issue5300() throws IOException, SQLException, InterruptedException {
+        File file = createInputScript("""
+                CREATE LINEAR AGGREGATE u64_sum(value INT) RETURNS INT;
+                
+                CREATE TABLE A (
+                    id VARCHAR(20) NOT NULL PRIMARY KEY,
+                    a VARCHAR(64),
+                    b VARCHAR(64),
+                    sno INT,
+                    small BIGINT,
+                    num1 INT,
+                    num2 INT
+                ) WITH ('append_only' = 'true');
+                
+                CREATE VIEW b AS
+                SELECT id, a, b, u64_sum(num1), u64_sum(num2), SUM(small), MAX(sno)
+                FROM A
+                GROUP BY id, a, b""");
+
+        File udf = Paths.get(RUST_DIRECTORY, "udf.rs").toFile();
+        PrintWriter script = new PrintWriter(udf, StandardCharsets.UTF_8);
+        script.println("""
+                use feldera_sqllib::*;
+                
+                pub type u64_sum_accumulator_type = i64;
+                
+                pub fn u64_sum_map(val: i32) -> u64_sum_accumulator_type {
+                    val.into()
+                }
+                
+                pub fn u64_sum_post(val: u64_sum_accumulator_type) -> i32 {
+                    val.try_into().unwrap()
+                }
+                """);
+        script.close();
+        CompilerMessages messages = CompilerMain.execute("-o", BaseSQLTests.TEST_FILE_PATH, file.getPath());
+        if (messages.errorCount() > 0)
+            throw new RuntimeException(messages.toString());
+        Utilities.compileAndTestRust(BaseSQLTests.RUST_DIRECTORY, false);
+
+        // Truncate file to 0 bytes
+        FileWriter writer = new FileWriter(udf);
+        writer.close();
+    }
+
+
+    @Test
     public void testUDA2() throws IOException, InterruptedException, SQLException {
         File file = createInputScript("""
                 CREATE LINEAR AGGREGATE I128_SUM(s BINARY(16)) RETURNS BINARY(16);

